apps/evm_family/evm_contracts.c

```diff
@@ -281,7 +281,13 @@ uint8_t ETH_ExtractArguments(const uint8_t *pAbiPayload,
   const uint8_t *pPayloadBasePtr = pCurrHeadPtr;
   uint8_t currArgument;
 
-  for (currArgument = 0; currArgument < numArgsInFunction; currArgument++) {
+  for (currArgument = 0; currArgument <= numArgsInFunction; currArgument++) {
+    // only return OK when completely parsed whole payload
+    if ((pAbiPayload + sizeOfPayload) == pCurrHeadPtr) {
+      returnCode = ETH_UTXN_ABI_DECODE_OK;
+      break;
+    }
+
```

Comment:
The if condition can be outside the loop when no arguments are there.

Reply:
changed